src/app/account/login/login.component.ts

```diff
@@ -58,6 +67,35 @@ export class LoginComponent implements OnInit, OnDestroy {
             this.errorMessage = LOGIN_ERROR_MESSAGES[error] || LOGIN_ERROR_MESSAGES.internalError;
         }
     }
+    onSubmit() {
+        const encryptedPassword = this.authService.encryptPassword(this.loginGroup.value.password);
+        const loginData = {
+            'email': this.loginGroup.value.email,
+            'password': encryptedPassword,
+          };
+        this.disableLoginBtn = true;
+
+        this.authService.basicAuthLogin(loginData)
+        .pipe(finalize(() => this.disableLoginBtn = false ))
+        .subscribe((response: any) => {
+            this.userStoreService.updateUserData(response.data);
+            this.router.navigateByUrl(this.returnURL);
+        },
+        (error: any) => {
+            if (error.status === 404) {
+                this.error(LOGIN_ERROR_TYPES.invalidEmailOrPassword);
```

Comment:
Please use a conditional statement here.
```suggestion
                this.error(error.status === 404 ? LOGIN_ERROR_TYPES.invalidEmailOrPassword : ...);
```

Reply:
done.